RUFAS/task_manager.py

```diff
@@ -180,6 +184,51 @@ def start(
             export_input_data_to_csv=export_input_data_to_csv,
         )
 
+    def get_rufas_version(self) -> str:
+        """
+        Returns the current version of RUFAS.
+
+        Returns
+        -------
+        str
+            Version of RUFAS or "Unknown" if the person is using a Python version earlier than 3.11.
+        """
+        user_python_version = Version(f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}")
+        if user_python_version < Version("3.11"):
+            return "Unknown"
+        with open(pyproject_file_path, "rb") as pyproject_file:
+            import tomllib
+
+            rufas_version = tomllib.load(pyproject_file)["project"]["version"]
+        return str(rufas_version)
+
+    def check_python_version(self) -> None:
+        """
+        Checks if the Python version meets minimum version set in pyproject.toml.
+        """
+        user_python_version = Version(f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}")
+        if user_python_version < Version("3.11"):
```

Comment:
```suggestion
        if user_python_version <= Version("3.11"):
```
People need to be on 3.12 at least right?

Also, might be good to make "3.11" or Version("3.11") a module constant

Reply:
Yea sorry this is confusing but this if statement is to catch a bad version that we can't catch by checking against the pyproject.toml file setting which we can't check without the tomllib library which isn't available in python earlier than 3.11.

The message in the add_error should be corrected to have the minimum required version be 3.12 though, good catch.